recursor build without protobuf support is failing

- Program: Recursor
 - Issue type: Bug report

### Short description
```
DEBUG: BUILDSTDERR: pdns_recursor.cc: In function 'void handleRunningTCPQuestion(int, FDMultiplexer::funcparam_t&)':
DEBUG: BUILDSTDERR: pdns_recursor.cc:1791:222: error: 'logQuery' was not declared in this scope
DEBUG: BUILDSTDERR:                  dc->d_tag = t_pdl->gettag_ffi(dc->d_source, dc->d_ednssubnet.source, dc->d_destination, qname, qtype, &dc->d_policyTags, dc->d_data, ednsOptions, true, requestorId, deviceId, dc->d_ttlCap, dc->d_variable, logQuery);
DEBUG: BUILDSTDERR:                                                                                                                                                                                                                               ^
DEBUG: BUILDSTDERR: pdns_recursor.cc: In function 'std::string* doProcessUDPQuestion(const string&, const ComboAddress&, const ComboAddress&, timeval, int)':
DEBUG: BUILDSTDERR: pdns_recursor.cc:1987:174: error: 'logQuery' was not declared in this scope
DEBUG: BUILDSTDERR:                ctag = t_pdl->gettag_ffi(source, ednssubnet.source, destination, qname, qtype, &policyTags, data, ednsOptions, false, requestorId, deviceId, ttlCap, variable, logQuery);
DEBUG: BUILDSTDERR:                                                                                                                                                                               ^
DEBUG: BUILDSTDERR: make[2]: *** [pdns_recursor.o] Error 1
DEBUG: BUILDSTDERR: make[2]: *** Waiting for unfinished jobs....
DEBUG: make[2]: Leaving directory `/builddir/build/BUILD/pdns-recursor-0.0.2253g4c928be'
DEBUG: BUILDSTDERR: make[1]: *** [all-recursive] Error 1
DEBUG: make[1]: Leaving directory `/builddir/build/BUILD/pdns-recursor-0.0.2253g4c928be'
DEBUG: BUILDSTDERR: make: *** [all] Error 2
```

### Environment
 - Software version: master  4c928be

--- a/pdns/pdns_recursor.cc
+++ b/pdns/pdns_recursor.cc
@@ -1766,12 +1766,13 @@ static void handleRunningTCPQuestion(int fd, FDMultiplexer::funcparam_t& var)
       bool needXPF = g_XPFAcl.match(conn->d_remote);
       string requestorId;
       string deviceId;
+      bool logQuery = false;
 #ifdef HAVE_PROTOBUF
       auto luaconfsLocal = g_luaconfs.getLocal();
       if (checkProtobufExport(luaconfsLocal)) {
         needECS = true;
       }
-      bool logQuery = t_protobufServer && luaconfsLocal->protobufExportConfig.logQueries;
+      logQuery = t_protobufServer && luaconfsLocal->protobufExportConfig.logQueries;
 #endif
 
       if(needECS || needXPF || (t_pdl && (t_pdl->d_gettag_ffi || t_pdl->d_gettag))) {
@@ -1932,6 +1933,7 @@ static string* doProcessUDPQuestion(const std::string& question, const ComboAddr
   ComboAddress destination = destaddr;
   string requestorId;
   string deviceId;
+  bool logQuery = false;
 #ifdef HAVE_PROTOBUF
   boost::uuids::uuid uniqueId;
   auto luaconfsLocal = g_luaconfs.getLocal();
@@ -1941,7 +1943,7 @@ static string* doProcessUDPQuestion(const std::string& question, const ComboAddr
   } else if (checkOutgoingProtobufExport(luaconfsLocal)) {
     uniqueId = (*t_uuidGenerator)();
   }
-  bool logQuery = t_protobufServer && luaconfsLocal->protobufExportConfig.logQueries;
+  logQuery = t_protobufServer && luaconfsLocal->protobufExportConfig.logQueries;
   bool logResponse = t_protobufServer && luaconfsLocal->protobufExportConfig.logResponses;
 #endif
   EDNSSubnetOpts ednssubnet;

--- a/pdns/rec-lua-conf.cc
+++ b/pdns/rec-lua-conf.cc
@@ -81,6 +81,7 @@ static void parseRPZParameters(const std::unordered_map<string,boost::variant<ui
   }
 }
 
+#if HAVE_PROTOBUF
 typedef std::unordered_map<std::string, boost::variant<bool, uint64_t, std::string> > protobufOptions_t;
 
 static void parseProtobufOptions(boost::optional<protobufOptions_t> vars, ProtobufExportConfig& config)
@@ -117,6 +118,7 @@ static void parseProtobufOptions(boost::optional<protobufOptions_t> vars, Protob
     config.logResponses = boost::get<bool>((*vars)["logResponses"]);
   }
 }
+#endif /* HAVE_PROTOBUF */
 
 void loadRecursorLuaConfig(const std::string& fname, luaConfigDelayedThreads& delayedThreads)
 {

